### ISSUE TITLE
Fee breakdown order note is broken

### ISSUE THREAD

**Issue by mehamasum** [Automattic/woocommerce-payments#7030]
### Describe the bug
Fee breakdown order notes are not being added correctly.

There is a `E_WARNING: Undefined array key "discount-"` error is produced when an order note is being added.
Stack trace:
```
 in WC_Payments_Captured_Event_Note::get_fee_breakdown called at /plugins/woocommerce-payments/includes/class-wc-payments-captured-event-note.php (142)
 in WC_Payments_Captured_Event_Note::compose_fee_break_down called at /plugins/woocommerce-payments/includes/class-wc-payments-captured-event-note.php (60)
 in WC_Payments_Captured_Event_Note::generate_html_note called at /plugins/woocommerce-payments/includes/class-wc-payments-order-service.php (403)
 in WC_Payments_Order_Service::add_fee_breakdown_to_order_notes called at /var/www/wp-includes/class-wp-hook.php (308)
```

### To Reproduce
Use an account where there is a discount applied to the fee.

### Actual behavior
The fee breakdown in order note shows up as:
![image](https://github.com/Automattic/woocommerce-payments/assets/12912469/c3f82a1f-2553-4a17-bf24-059fddaa18a2)

Whereas, the actual breakdown in transaction details page shows the correct breakdown:
![image](https://github.com/Automattic/woocommerce-payments/assets/12912469/43aa4c24-1ed0-49c8-9c9a-588360e70c5e)

### Expected behavior
The order note should have the same breakdown as the transaction details page.

### Additional context
I did some digging and believe this could be happening because [in here](https://github.com/Automattic/woocommerce-payments/blob/900951661b877a04f8643596e1adf2fa1b69badf/includes/class-wc-payments-captured-event-note.php#L209-L211), when `$fee['additional_type']` is set, we concat it with `$label_type` by a hyphen.

Since there is only check for `isset` and for some reason `$fee['additional_type']` is an empty string, the empty value will get concatinated with the `$label_type`.

I tried changing the `isset( $fee['additional_type'] )` to `! empty( $fee['additional_type']` which stopped the error from coming in and also seems to fix the above issue. But you may need to check whether this should be a client side fix or a server side one.

**Comment by RadoslavGeorgiev** [comment]
This issue fees, so assigning to @Automattic/pulsar  (based on team responsibilities Pc2DNy-3z-p2) cc @kalessil as part of Gamma Triage process PcreKM-yM-p2.

**Comment by kalessil** [comment]
Thanks @RadoslavGeorgiev, adding to our Sprints backlog.

**Comment by mgascam** [comment]
I'm not able to reproduce the issue in my dev environment. I'm adding a discount fee, following the step to reproduce in the description. This is how the order note looks like for me:

<img width="258" alt="Screenshot 2023-11-07 at 16 07 51" src="https://github.com/Automattic/woocommerce-payments/assets/1553182/ae23b673-267c-452a-8953-7c5919cfaeed">

This bug was fixed when working on another error in the order notes https://github.com/Automattic/woocommerce-payments/pull/7024. It was released in [v6.4.0](https://github.com/Automattic/woocommerce-payments/releases/tag/6.4.0).
